zvariant/src/signature.rs

```diff
@@ -288,6 +293,18 @@ impl<'a> Signature<'a> {
 
         clone
     }
+
+    /// Returns the number of complete types for the signature.
+    ///
+    /// **Note**: if the signature is invalid, returns the first error.
```

Comment:
Let's keep consistency with Rust's conventions and mention we mean top-level types:

```suggestion
    /// The number of complete top-level signature in `self`.
    ///
    /// # Errors
    /// 
    /// If the signature is invalid, returns the first error.
```

Reply:
Hmm? you confused me even more. I would assume a Rust "top-level type" is one exposed at the top-level of a namespace... Let's follow the D-Bus spec here. Do not mix Rust types with D-Bus.

Reply:
I'm not confusing them. I'm talking about D-Bus types. If this method returns 2 for `uas`, that means it's only counting the top-level signatures, no?

Reply:
This is not the spec jargon. it's called "complete types"

Reply:
ok. What about rest of the changes suggested for the doc comment?